lang/semgrep-grammars/src/semgrep-c-sharp/grammar.js

```diff
@@ -86,6 +86,10 @@ module.exports = grammar(standard_grammar, {
         previous,
         prec.right(100, seq($.ellipsis, ';')),  // expression ellipsis
         prec.right(100, $.ellipsis),  // statement ellipsis
+        seq($.deep_ellipsis, ';'),
```

Comment:
Why do we need that? deep_ellipsis is part of an expression no? so it should work without this extra case.

Reply:
It is, but the C# grammar has refined it so that `seq(expression, ';')` is no longer permitted, but instead an explicit `expression_statement_expression`, which outlines the specific expressions which are allowed to appear before a semicolon.

A general expression is not one of those, so deep ellipsis also becomes one of those.

Reply:
Can you add a comment in the code explaining why we need this duplication?
Otherwise in 6months we will have forgotten.